lib/cached_network_image.dart

```diff
@@ -215,57 +315,94 @@ class CacheObject {
     return null;
   }
 
+  DateTime touched;
+  String url;
+
   Object lock;
   Map _map;
 
-  CacheObject() {
+  CacheObject(String url) {
+    this.url = url;
     _map = new Map();
+    touch();
     lock = new Object();
   }
 
-  CacheObject.fromMap(Map map) {
+  CacheObject.fromMap(String url, Map map) {
+    this.url = url;
     _map = map;
+
+    if (_map.containsKey("touched")) {
+      touched = new DateTime.fromMillisecondsSinceEpoch(_map["touched"]);
+    } else {
+      touch();
+    }
+
     lock = new Object();
   }
 
   Map toMap() {
     return _map;
   }
 
+  touch() {
+    touched = new DateTime.now();
+    _map["touched"] = touched.millisecondsSinceEpoch;
+  }
+
   setDataFromHeaders(Map<String, String> headers) async {
```

Comment:
`setDataFromHeaders` is very coupled with HttpHeaders. Cache Objects is not support to know about headers - it knows the object, knows when it was touched etc... 
parsing HttpHeaders sounds like an infrastructure concern? Currently Cache Manager knows that files are being downloaded from Http so it shall parse HttpHeaders (or delegate it to some parser etc..)

Reply:
That is probably a good idea.